src/main/java/luke/TaskList.java

```diff
@@ -0,0 +1,112 @@
+package luke;
+
+import java.util.ArrayList;
+
+//for CURRENT tasks only.
+public class TaskList {
+
+    private ArrayList<Task> tasks = new ArrayList<Task>();
+
+    //fetch tasks from history
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Adds task to tasklist.
+     *
+     * @param task The task to be added.
+     */
+    public void addTask(Task task) {
+        if (task != null) {
+            tasks.add(task);
+            System.out.println("I helped you add task '" + task.fullStatus() + "'. But do it yourself next time! Hmmph!" + "\n");
+        }
+    }
+
+    /**
+     * Marks a task at index in the tasklist as done.
+     *
+     * @param index Index of task to be marked as done.
+     */
+    public void markTask(int index) {
+        try {
+            tasks.get(index).complete();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Jeez, you really ought to give me a number I can work with... got that?");
+            System.out.println("[Item index exceeds history count]\n");
+            return;
```

Comment:
Maybe return a boolean. True/ false representing whether the task was successfully executed. Instead of having an empty return.